File: script/gibbs.m
A=2; T=8; tau=4; w=2*pi/T;
x=linspace(-5,5,500);
y=A*(heaviside(x+2)-heaviside(x-2));
figure(1);
subplot(221);
plot(linspace(-4,4,500),y,'linewidth',1); grid on; xlim([-5,5]); ylim([-0.5,2.5]); 
xlabel('$t$','interpreter', 'latex'); ylabel('$f(t)$','interpreter', 'latex');
y=fc(x,3);
subplot(222);
plot(x,y,'linewidth',1); axis tight; grid on; ylim([-0.5,2.5]); 
xlabel('$t$','interpreter', 'latex'); ylabel('$S_3[f](t)$', 'interpreter', 'latex'); title('n=3');
y=fc(x,15);
subplot(223);
plot(x,y,'linewidth',1); axis tight; grid on; ylim([-0.5,2.5]); 
xlabel('$t$','interpreter', 'latex'); ylabel('$S_{15}[f](t)$', 'interpreter', 'latex'); title('n=15');
subplot(224);
y=fc(x,70);
plot(x,y,'linewidth',1); axis tight; grid on; ylim([-0.5,2.5]); 
xlabel('$t$','interpreter', 'latex'); ylabel('$S_{70}[f](t)$', 'interpreter', 'latex'); title('n=70');



function y=fc(x,n)
    y=zeros(1,500);
    for m=-n:n
        y=y+sinc(m/2)*exp(1i*m*pi/4*x);
    end
end

File: script/gibbs.py
import numpy as np
import matplotlib.pyplot as plt

def fc(x,n):
    y=np.ones(500)
    for m in range(1,n+1):
        y = y + 2*np.sinc(m/2)*np.cos(np.pi*x/4*m)
    return y

def plotapprox(subplot, n):
    plt.subplot(subplot)
    y=fc(x,n)
    plt.plot(x,y)
    plt.grid(axis='both',linewidth=0.5)
    plt.xlim((-5,5))
    plt.ylabel('S[f](t)')
    plt.title('n={}'.format(n))

x=np.linspace(-5,5,500)
plt.subplot(221)
y=2*(np.heaviside(x+2,0.5)-np.heaviside(x-2,0.5))
plt.plot(np.linspace(-4,4,400),y[50:450])
plt.grid(axis='both',linewidth=0.5)
plt.xlim((-5,5))
plt.ylabel('f(t)')

plotapprox(222,3)
plotapprox(223,15)
plotapprox(224,100)

plt.show()

